guides/v2.1/cloud/cdn/cloud-vcl-custom-snippets.md

```diff
@@ -11,223 +11,324 @@ functional_areas:
+
+-   [Regular snippets](https://docs.fastly.com/guides/vcl-snippets/using-regular-vcl-snippets)—Versioned VCL snippets with VCL code and settings associated with a specific version of the VCL code for your environment. You create, modify, and deploy regular VCL snippets to the Fastly service configuration for a specific environment.
+
+-   [Dynamic snippets](https://docs.fastly.com/guides/vcl-snippets/using-dynamic-vcl-snippets)—Unversioned VCL snippets created using the Fastly API. You can modify these snippets and deploy them without having to update the Fastly VCL version for your service.
+
+We recommend using custom VCL snippets with Edge Dictionaries and Access Control Lists (ACL) to store data used in your custom code. 
+
+-  [**Edge dictionary**](https://docs.fastly.com/guides/edge-dictionaries/about-edge-dictionaries)—Stores data as key-value pairs in a dictionary container that can be referenced from custom VCL snippets
+
+-  [**Edge ACL**](https://docs.fastly.com/guides/access-control-lists/about-acls)—Stores the client IP address data that defines the access control list for block or allow rules implemented using custom VCL snippets
+
+
+The dictionary and ACL data is deployed to the Fastly Edge nodes accessible across network regions. Additionally, the data can be updated dynamically across the network without requiring you to redeploy the VCL code for your staging or production environment.
+
+{: .bs-callout .bs-callout-info}
+You must [set up Fastly]({{ page.baseurl }}/cloud/cdn/configure-fastly.html) before you can add custom VCL snippets.
+
+## Custom VCL snippet examples and tutorials
+
+The examples and instructions included in the {{ site.data.var.ece }} documentation explain how to use regular custom VCL snippets with edge dictionaries and edge ACLs.  It is not a comprehensive guide to Fastly VCL. For detailed VCL tutorial, reference, and troubleshooting information, see the following documentation:
+
+-   [Guide to Fastly VCL](https://docs.fastly.com/guides/vcl/guide-to-vcl)—High level information about the Fastly Varnish implementation, Fastly VCL extensions, and resources for learning more about Varnish and VCL.
+-   [Fastly VCL reference](https://docs.fastly.com/guides/vcl/)—Detailed programming reference to develop and troubleshoot Fastly custom VCL and custom VCL snippets.
+
+
+You can create and manage custom VCL snippets from the Magento Admin UI or by using the Fastly API:
+
+- [Magento Admin UI](#manage-custom-vcl-snippets-from-the-admin-ui)—We recommend using the Magento Admin UI to create, add, and manage VCL snippets because it automates the process of cloning the active VCL code (VCL version) for editing, saving and validating the custom VCL snippet, inserting the snippet into the VCL version, and uploading the new VCL version to Fastly. Additionally, you can create, view, and manage snippets from the UI without submitting multiple API requests and manually managing the active VCL version.
```

Comment:
```suggestion
- [Magento Admin UI](#manage-vcl-snippets-from-the-admin-ui)—We recommend using the Magento Admin UI to create, add, and manage VCL snippets because it automates the process of cloning the active VCL code (VCL version) for editing, saving and validating the custom VCL snippet, inserting the snippet into the VCL version, and uploading the new VCL version to Fastly. Additionally, you can create, view, and manage snippets from the UI without submitting multiple API requests and manually managing the active VCL version.
```